manifests/cirunner.pp

```diff
@@ -165,14 +147,53 @@
     refreshonly => true,
     require     => Package[$package_name],
   }
-
-  $runners_hash = hiera_hash($hiera_runners_key, {})
-  $runners = keys($runners_hash)
-  $default_config = hiera_hash($hiera_default_config_key, {})
-  gitlab::runner { $runners:
-    binary         => $package_name,
-    default_config => $default_config,
-    runners_hash   => $runners_hash,
-    require        => Exec['gitlab-runner-restart'],
+  if ! empty($docker_runners) {
+    if defined(Class['gitlab::cirunner::docker']) {
+      warning('Class[\'gitlab::cirunner::docker\' is defined so $gitlab::cirunner::docker__runners willbe ignored')
+    } else {
+      class {'::gitlab::cirunner::docker': runners => $docker_runners}
```

Comment:
you don't need the :: in front of the class names

Reply:
done